### PR TITLE
Fix: remove redundant distinct with views

### PR DESCRIPTION
**Pull request by devcrafter**
Redundant DISTINCT wasn't removed when querying from a view which already contains the same DISTINCT. Example:
```sql
create view tab_v as select distinct(x) from tab;
select distinct(x) from tab_v;
```

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 6c8d5ca0a554ecc4fee32269858797d139f3c02a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52438/6c8d5ca0a554ecc4fee32269858797d139f3c02a/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by devcrafter** [comment]
For this example - https://pastila.nl/?0063fdd0/fb948ad1b33ab14fc767bbff6a8053c1
Merged DAG (all DAGs between 2 DistinctStep), which is used to find the original column name, looks like this - https://pastila.nl/?1a59d7a5/06bf5a480da41e46e207f1a15bd4528d (it contains `materialized` function apart from aliases)
So, the search is failed here https://github.com/ClickHouse/ClickHouse/blob/69b897d726139fe1732ea6be9d5bc46a4022c730/src/Processors/QueryPlan/Optimizations/removeRedundantDistinct.cpp#L249-L251

File: src/Interpreters/ActionsDAG.cpp [hunk 1]
```diff
@@ -2511,11 +2511,21 @@ FindOriginalNodeForOutputName::FindOriginalNodeForOutputName(const ActionsDAGPtr
...
-            node = node->children.front();
+            if (node->type == ActionsDAG::ActionType::ALIAS)
+            {
+                node = node->children.front();
+            }
+            /// materiailze() function can occur when dealing with views
+            /// TODO: not sure if it should be done here, looks too generic place
+            else if (node->type == ActionsDAG::ActionType::FUNCTION && node->function_base->getName() == "materialize")
+            {
+                chassert(node->children.size() == 1);
```
Comment (alexey-milovidov) [inline src/Interpreters/ActionsDAG.cpp:2524]: If a user writes `materialize(a)(b)`, will it be checked earlier?

Comment (devcrafter) [reply src/Interpreters/ActionsDAG.cpp:2524]: Sorry, don't get your question. But added explicitly query into the test with `materialize()` to check explicitly that usage of `materialize()` doesn't affect the optimization.

> If a user writes materialize(a)(b), will it be checked earlier?
(1) how `materialize()` can have 2 parameters?
(2) what do you mean by check earlier? Here we check it while walking through `ActionsDAG` for all expressions which are between preliminary `DistinctStep` (redundant) and final `DistinctStep`(first distinct from data source)

**Pull request merged**